src/fetchers/default-fetcher.nix

```
@@ -11,6 +11,7 @@
   defaultPackageVersion,
   sourceOverrides,
   sources,
+  sourceRoot,
```

This should be optional rather than required.

```suggestion
  sourceRoot ? null,
```
an error should be thrown if `sourceRoot` needs to be used but it is `null`.

---

I wanted to make it optional at first, but then I thought I've seen something downstream that has it required, so I figured it's better to make it required as well.

But maybe I confused it with a single package's source. I'll adjust it how you suggest.

---

These should be optional mostly because this argument didn't exist before (to prevent code from breaking). And a `sourceRoot` may not always exist (for example, when building from a dream-lock directly).